### ISSUE TITLE
Illegal division by zero; creating unique scaffold links failed; Assembly with flye failed

### ISSUE THREAD

**Issue by anoukwillemsen** [alekseyzimin/masurca#243]
Hello,

I am running MaSuRCA for performing hybrid assemblies with PromethION and Illumina data. I am using Flye for final assembly of the corrected mega-reads. 

The run finishes but in the MaSuRCA STDOUT (see attached "slurm-1710475.out") I get the following errors:
Illegal division by zero at /apps/masurca/4.0.4/bin/find_repeats.pl line 43, <FILE> line 551.
[Mon Jul 12 14:28:10 CEST 2021] creating unique scaffold links failed
[Mon Jul 12 14:28:10 CEST 2021] Assembly with flye failed, see files under flye/

However, when looking at the flye.log (also attached) it seems that flye finishes normally and the final assembly is available in the flye folder. Does this mean that the run did finish correctly despite of the above mentioned errors? Or does this mean that this is an intermediate assembly (which I suspect because of the folder name: flye.mr.41.17.15.0.02/)

Best,

Anouk

[flye.log.txt](https://github.com/alekseyzimin/masurca/files/6823398/flye.log.txt)
[slurm-1710475.out.txt](https://github.com/alekseyzimin/masurca/files/6823399/slurm-1710475.out.txt)

**Comment by alekseyzimin** [comment]
Yes, Flye finished but post-Flye scaffolding did not.  You can use assembly.fasta output by Flye.  This is fixed in the next version 4.0.5.
